src/Raven.Server/Documents/Handlers/DocumentHandler.cs

```diff
@@ -551,24 +554,57 @@ public class MergedPutCommand : TransactionOperationsMerger.MergedTransactionCom
 
         public ExceptionDispatchInfo ExceptionDispatchInfo;
         public DocumentsStorage.PutOperationResults PutResult;
+        private bool _validateAttachments;
 
         public static string GenerateNonConflictingId(DocumentDatabase database, string prefix)
         {
             return prefix + database.DocumentsStorage.GenerateNextEtag().ToString("D19") + "-" + Guid.NewGuid().ToBase64Unpadded();
         }
 
-        public MergedPutCommand(BlittableJsonReaderObject doc, string id, LazyStringValue changeVector, DocumentDatabase database)
+        public MergedPutCommand(BlittableJsonReaderObject doc, string id, LazyStringValue changeVector, DocumentDatabase database, bool validateAttachments = false)
         {
             _document = doc;
             _id = id;
             _expectedChangeVector = changeVector;
             _database = database;
+            _validateAttachments = validateAttachments;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void ValidateAttachmentHash(DocumentsOperationContext context, string id, BlittableJsonReaderObject document)
+        {
+            if (document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata)
+                && metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray attachments))
+            {
+                ValidateAttachmentHashInternal(context, id, attachments);
+            }
+        }
+
+        private void ValidateAttachmentHashInternal(DocumentsOperationContext context, string id, BlittableJsonReaderArray attachments)
+        {
```

Comment:
Just in case can we check if attachments array is not null? I think that `{ '@attachments' : null }` might be a problem here?